docs/developer/architecture/configurations.rst

```diff
@@ -73,3 +73,197 @@ As a MetalK8s expert, I can use ``kubectl`` command(s) in order to edit all
+CRUD operations on all KV data stored in it.
+Consul KV also allows access to past versions of objects and has an optimistic
+concurrency when manipulating multiple objects.
+
+Note that, Consul KV store was rejected because managing operations such as
+performing full backups, system restores for a full fledged KV system
+requires time and much more efforts than the ConfigMap approach.
+
+Operator (Custom Controller) vs Salt
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Operators are useful in that, they provide self-healing functionalities on a
+reactive basis. When a user changes a given configuration, it is easy to
+reconcile and apply these changes to the in-cluster objects.
+
+The Operator approach was rejected because it is much more complex, requires
+much more effort to realize and there is no real need for applying changes
+using this method because configuration changes are not frequent
+(for a typical MetalK8s admin, changing the number of replicas for a given
+deployment could happen once in 3 months or less) as such, having an operator
+watch for object changes is not significant and not very useful at this point
+in time.
+
+
+Implementation Details
+----------------------
+
+Iteration 1
+^^^^^^^^^^^
+
+- Define and deploy new ConfigMap stores that will hold cluster and service
+  configurations as listed in the requirements. For each ConfigMap, define its
+  schema, its default values, and how it impacts the configured services
+- Template and render Deployment and Pod manifests that will make use of
+  this persisted cluster and service configurations
+- Document how to change cluster and service configurations using kubectl
+- Document the entire list of configurations which can be changed by the user
+
+Iteration 2
+^^^^^^^^^^^
+
+- Provide a CLI tool for changing any of the cluster and service
+  configurations:
+
+    - Count of replicas for chosen Deployments (Prometheus)
+    - Updating a Dex authentication connector (OpenLDAP, AD and
+      staticUser store)
+    - Updating the Alertmanager notification configuration
+
+- Provide a UI interface for adding, updating and deleting service specific
+  configurations for example Dex-LDAP connector integration.
+- Provide a UI interface for listing MetalK8s available/supported Dex
+  authentication Connectors
+- Provide a UI interface for enabling or disabling Dex authentication
+  connectors (LDAP, Active Directory, StaticUser store)
+- Add a UI interface for listing Alertmanager notification systems MetalK8s
+  will support (Slack, email)
+- Provide a UI interface for adding, modifying and deleting Alertmanager
+  configurations from the listing above
+
+Documentation
+-------------
+
+In the Operational Guide:
+
+* Document how to customize or change any given service settings using the CLI
+  tool
+* Document how to customize or change any given service settings using the UI
+  interface
+* Document the list of service settings which can be configured by the user
+
+Test Plan
```

Comment:
Shouldn't you check that an upgrade of MetalK8s doesn't break previous customizations?

Reply:
Indeed, we should